scss/civicrm/common/_block-shadows.scss

```diff
@@ -0,0 +1,55 @@
+/* stylelint-disable max-nesting-depth, selector-max-compound-selectors, selector-no-qualifying-type, selector-max-id */
+
+/**
+ * Applies proper shadows to the <h3> + <div class="crm-block"> pair.
+ *
+ * @NOTE
+ * If you apply shadows to these elements directly (not via :before pseudoclass),
+ * then the shadows will overlap. The method makes it visually look like there is
+ * just a single shadow underneath the pair.
+ */
+
+// Need that otherwise shadows will be cropped on the top and the bottom
+#crm-main-content-wrapper {
+  position: relative;
+  z-index: 0;
+}
+
+*:not(.crm-form-block) {
+  > h3:not(.crm-severity-info) {
+    &,
+    + .crm-block {
+      // Discard direct shadows
+      box-shadow: none !important;
+      position: relative;
+
+      &::before {
+        bottom: 0;
+        box-shadow: $box-shadow;
+        content: '';
+        left: 0;
+        position: absolute;
+        right: 0;
+        top: 0;
+        // This is needed to place the shadows underneath so they do not overlap
+        z-index: -1;
+      }
+    }
+  }
+}
+
+h3::before {
+  border-radius: #{$border-radius-base} #{$border-radius-base} 0 0 !important;
+}
+
+h3 + .crm-form-block {
+  &,
+  &::before {
+    border-radius: 0 0 #{$border-radius-base} #{$border-radius-base} !important;
+  }
+
+  .crm-submit-buttons {
+    // Need to discard the background otherwise this block will have sharp edges
+    background-color: transparent !important;
```

Comment:
are all these rules defined with `!important` in CiviCRM?

Reply:
If we don't use !important we will have to write a completely unreadable chain of elements, plus I want to be 100% sure this exact block never has a background.